crates/router/src/core/payments.rs

```diff
@@ -760,47 +745,148 @@ pub async fn add_process_sync_task(
     Ok(())
 }
 
-pub async fn route_connector<F>(
+pub fn update_straight_through_routing<F>(
+    payment_data: &mut PaymentData<F>,
+    request_straight_through: serde_json::Value,
+) -> CustomResult<(), errors::ParsingError>
+where
+    F: Send + Clone,
+{
+    let mut routing_data: storage::RoutingData = payment_data
+        .payment_attempt
+        .connector
```

Comment:
If the connector field is empty in payment_attempt, then this unwrap will return `{}` which will not be parsed in the next step. Is this intended behaviour?

Reply:
The type it's being parsed to has all fields as optional, so it will parse to the type with all values as null. This is intended behavior.